python/Ganga/GPIDev/Base/Objects.py

```diff
@@ -771,19 +715,16 @@ def __init__(cls, name, bases, this_dict):
         # create reference in schema to the pluginclass
         this_schema._pluginclass = cls
 
-        # store generated proxy class
-        setattr(cls, '_proxyClass', proxyClass)
-
-        # register plugin class
-        if hasattr(cls, '_declared_property'):
-            # if we've not even declared this we don't want to use it!
-            if not cls._declared_property('hidden') or cls._declared_property('enable_plugin'):
-                allPlugins.add(cls, cls._category, getName(cls))
+        # if we've not even declared this we don't want to use it!
+        if not cls._declared_property('hidden') or cls._declared_property('enable_plugin'):
+            allPlugins.add(cls, cls._category, getName(cls))
 
-            # create a configuration unit for default values of object properties
-            if not cls._declared_property('hidden') or cls._declared_property('enable_config'):
-                this_schema.createDefaultConfig()
+        # create a configuration unit for default values of object properties
+        if not cls._declared_property('hidden') or cls._declared_property('enable_config'):
+            this_schema.createDefaultConfig()
 
+        # store generated proxy class
+        setattr(cls, '_proxyClass', GPIProxyClassFactory(name, cls))
```

Comment:
If we decide that we want _all_ `GangaObject`s to be exported (which we currently don't since we set `_hidden` on many) then I agree but I think it's good to be explicit with the GPI where possible. For example we don't want `GangaObject` itself exported or any of the adapters or one or two other examples.

The way it works at the moment is based on `_hidden` and adding it to `allPlugins` and then the bootstrap iterating that and adding each to the GPI.